### PR TITLE
Add site plugin methods

### PR DESCRIPTION
**Pull request by retrofox**
### Updated/added methods

* Site#plugins()
* Site#plugin()

* SitePlugin#get()
* SitePlugin#config()
* SitePlugin#update()
* SitePlugin#install()
* SitePlugin#delete()
* SitePlugin#activate()
* SitePlugin#deactivate()
* SitePlugin#enableAutoupdate()
* SitePlugin#disableAutoupdate()

Also in this PR I've done some minor improvements such as sort some tests by name. :-o It isn't too importante but they are shown like big changes in this PR.

### CODE REVIEW

File: lib/site.js [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
  */
-import Post from './site.post';
 import Category from './site.category';
-import Tag from './site.tag';
-import Media from './site.media';
 import Comment from './site.comment';
-import SiteWordAds from './site.wordads';
 import Follow from './site.follow';
-import SitePlugin from './site.plugin';
+import Media from './site.media';
```
Comment (timmyc) [inline lib/site.js]: Since these are being re-ordered should we update `Media` -> `SiteMedia`?

Comment (retrofox) [inline lib/site.js]: Yes, but maybe in another PRs? I just wanted sort the current ones by name. I can create a specific PR for that.

File: lib/site.plugin.js [hunk 2]
```diff
@@ -1,38 +1,134 @@
...
 		}
 
-		this._id = id;
+		this._slug = encodeURIComponent( slug );
 		this._sid = sid;
-		this.path = `${root}/${this._sid}/plugins`;
 		this.wpcom = wpcom;
+
+		this.path = `${root}/${ this._sid }/plugins`;
+		this.pluginPath = `${ this.path }/${ this._slug }`;
```
Comment (timmyc) [inline lib/site.plugin.js]: Seems like since only `this.pluginPath` is used below, these two lines could be combined to create `this.pluginPath` here.

File: lib/site.plugin.js [hunk 3]
```diff
@@ -1,38 +1,134 @@
...
 	 */
-	constructor( id, sid, wpcom ) {
+	constructor( slug, sid, wpcom ) {
 		if ( ! ( this instanceof SitePlugin ) ) {
-			return new SitePlugin( id, sid, wpcom );
+			return new SitePlugin( slug, sid, wpcom );
 		}
 
-		this._id = id;
+		this._slug = encodeURIComponent( slug );
```
Comment (timmyc) [inline lib/site.plugin.js]: Since everything in this file is dependent upon the presence of `this._slug` should we add a check here to ensure that a value was passed in properly?

**Comment by timmyc** [comment]
So the diff in `test/test.wpcom.site.js` is a little hard to follow.  Did you just reorder the tests?

**Comment by retrofox** [comment]
> So the diff in test/test.wpcom.site.js is a little hard to follow. Did you just reorder the tests?

just re-order but let me get out those changes from here. I'll create a new PR just for that.

File: lib/site.plugin.js (hunk 3 above)
Comment (retrofox) [inline lib/site.plugin.js]: :+1:

File: lib/site.plugin.js (hunk 2 above)
Comment (retrofox) [inline lib/site.plugin.js]: :+1:

**Comment by timmyc** [comment]
@retrofox were you going to add some tests for `SitePlugin`?  One minor comment from before too.

File: lib/site.plugin.js [hunk 4]
```diff
@@ -1,38 +1,138 @@
...
+			throw new Error( '`slug` is not correctly defined' );
 		}
 
-		this._id = id;
+		this._slug = encodeURIComponent( slug );
 		this._sid = sid;
-		this.path = `${root}/${this._sid}/plugins`;
 		this.wpcom = wpcom;
+
+		const path = `${root}/${ this._sid }/plugins`;
```
Comment (timmyc) [inline lib/site.plugin.js]: I think I mentioned it before but it got lost in a force push, what do you think about combining this with the next line since it is only used there.

Comment (retrofox) [inline lib/site.plugin.js]: I did it in my way :-o
I replaced `this.path` by `const path`. Let me remove it completely.

Comment (timmyc) [inline lib/site.plugin.js]: Just a suggestion, you don't have to change anything :)

**Comment by timmyc** [comment]
@retrofox LGTM - I think we both need to do some document PRs to follow up as well.

File: lib/site.plugin.js (hunk 4 above)
Comment (retrofox) [inline lib/site.plugin.js]: :-)

**Comment by retrofox** [comment]
> @retrofox LGTM - I think we both need to do some document PRs to follow up as well.

yup, soon ;-)
